src/PendingReason.php

```diff
@@ -295,4 +356,58 @@ public function cleanDBonPurge()
             ]
         );
     }
+
+    public static function getDefault()
+    {
+        $pending_reason = new PendingReason();
+        if (
+            $pending_reason->getFromDBByCrit([
+                'is_default' => 1,
+            ])
+        ) {
+            return $pending_reason;
+        }
+
+        return null;
+    }
+
+    public static function isDefaultPending()
+    {
+        $default_pending = self::getDefault();
+
+        return $default_pending && $default_pending->fields['is_pending_per_default'];
+    }
+
+    public function prepareInput(array $input)
+    {
+        if (isset($input['is_default']) && $input['is_default']) {
+            global $DB;
+
+            $DB->update(
+                $this->getTable(),
+                [
+                    'is_default' => 0,
+                ],
+                [
+                    'is_default' => 1,
+                ]
+            );
```

Comment:
Updating directly in DB prevents hooks to be triggered.

```suggestion
            $previous_default = self::getDefault();
            if ($previous_default !== null) {
                 $previous_default->update(['id' => $previous_default->getId()] + ['is_default' => 0]);
            }
```